Exercise-02/Exercise02/main.swift

```
@@ -53,17 +71,399 @@ func runApplication() {
+    printInvalidCommand()
+    return nil
+  }
+  return insertedDate
+}
+
+func readHour() -> Int? {
+  print("Insert the hour of the day (from 0 to 23):")
+  guard let hourString = readLine(), !hourString.isEmpty, let insertedHour = Int(hourString) else {
+    printInvalidCommand()
+    return nil
+  }
+  
+  if( insertedHour < 0 || insertedHour > 23) {
+    printInvalidCommand()
+    return nil
+  }
+  return insertedHour
+}
+
+func readMeasure() -> Character? {
+  print("Insert the unit of measure (C, K or F):")
+  guard let insertedMeasure = readLine(), !insertedMeasure.isEmpty, insertedMeasure.count == 1 else {
+    printInvalidCommand()
+    return nil
+  }
+  return Character(insertedMeasure)
+}
+
+func readTemperature() -> Measurement<UnitTemperature>? {
+  let leftBoundary = Measurement(value: -50, unit: UnitTemperature.celsius)
+  let rightBoundary = Measurement(value: 50, unit: UnitTemperature.celsius)
+  
+  print("Insert the temperature:")
+  guard let temperatureString = readLine(), !temperatureString.isEmpty,
+    let insertedTemperature = Double(temperatureString) else {
+    printInvalidCommand()
+    return nil
+  }
+  
+  let insertedMeasure = readMeasure()
+  
+  var temperature: Measurement<UnitTemperature>
+  switch insertedMeasure {
+  case "K":
+    let kelvin = Measurement(value: insertedTemperature, unit: UnitTemperature.kelvin)
+    temperature = kelvin.converted(to: .celsius)
+  case "F":
+    let fahrenheit = Measurement(value: insertedTemperature, unit: UnitTemperature.fahrenheit)
+    temperature = fahrenheit.converted(to: .celsius)
+  case "C":
+    temperature = Measurement(value: insertedTemperature, unit: UnitTemperature.celsius)
+  default:
+    printInvalidCommand()
+    return nil
+  }
+  
+  if temperature <= leftBoundary || temperature >= rightBoundary {
+    print("The inserted temperature is too low or too high!")
```

you can throw an error here. You can also separate the two cases.